users/views.py

```diff
@@ -36,3 +38,55 @@ def post(self, request):
             
         except KeyError:
             return JsonResponse({'MESSAGE' : 'KEY_ERROR'}, status=400) 
+
+
+class ProfileRegisterView(View):
+    @login_decorator
+    def post(self, request):
+        s3_client = boto3.client(
+            's3',
+            aws_access_key_id=AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=AWS_SECRET_ACCESS_KEY
+        )
+        try:
+            user       = request.user
+            nickname   = request.POST.get('json')
+            image_file = request.FILES.get('file')
+            
+            key      = str(user.user_code)
+            location = s3_client.get_bucket_location(Bucket=AWS_S3_BUCKET_NAME)['LocationConstraint']
+                                
+            s3_client.upload_fileobj(
+                                image_file, 
+                                AWS_S3_BUCKET_NAME,
+                                f'user-profile-image/{key}/{image_file.name}',
+                                ExtraArgs={
+                                    "ContentType": image_file.content_type
+                                }
+                            )
+            
+            url      = f'http://{AWS_S3_BUCKET_NAME}.s3.{location}.amazonaws.com/user-profile-image/{key}/{image_file.name}'
+            
+            User.objects.filter(user_code=user.user_code).update(nickname=nickname, profile_image_url=url)
+            
+            return JsonResponse({'MESSAGE' : 'SUCCESS'}, status=200)
+        
+        except KeyError:
+            return JsonResponse({'MESSAGE' : 'KEY_ERROR'}, status=400)
+
+    @login_decorator
+    def get(self, request):
+        try:
+            user = request.user
+        
+            user_profile = {
+                "nickname"          : user.nickname,
+                "profile_image_url" : user.profile_image_url if user.profile_image_url != None else DEFAULT_IMAGE_URL
```

Comment:
조건문에서는 해당 조건의 boolean 값이 Truthy, Falsy 하냐를 판단할 수 있기때문에 변수를 그대로 조건으로 둘 수 있습니다.
```suggestion
                "profile_image_url" : user.profile_image_url if user.profile_image_url else DEFAULT_IMAGE_URL
```

Reply:
앗 그러네요! 수정하겠습니다.